Tools/test_shard_filter.py

```
@@ -337,50 +373,48 @@ def cmd_generate():
     path = sys.argv[2]
     if not os.path.exists(path):
         return
-    with open(path) as f:
-        content = f.read().strip()
-    if content:
-        methods = [part.replace("method==", "").strip("' ") for part in content.split("||")]
-        print(f"Running {len(methods)} test groups:", file=sys.stderr)
-        for method in methods:
-            print(f"  - {method}", file=sys.stderr)
-        print(content)
+    root = ET.parse(path).getroot()
```

Ruff (0.15.20)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)